### PR TITLE
Ignore benign error in node reconciler

### PR DESCRIPTION
**Pull request by redhatrises**
- Ignore benign error if an update has an out-of-date resource version.

### CODE REVIEW

**Review by isimluk — COMMENTED** [review]
I am trying to test this and I am getting:
```
1.6566725897840126e+09  INFO    Waiting for cleanup pods to complete. Retrying....      {"controller": "falconnodesensor", "controllerGroup": "falcon.crowdstrike.com", "controllerKind": "FalconNodeSensor", "falconNodeSensor": {"name":"default"}, "namespace": "", "name": "default", "reconcileID": "c675d42a-c41c-4454-b9f1-5ec989b3e112", "DaemonSet": "/default"}
```
in an infinite loop. Is that something that can be related to this PR?

File: controllers/falcon/falconnodesensor_controller.go [hunk 1]
```diff
@@ -765,6 +770,11 @@ func (r *FalconNodeSensorReconciler) conditionsUpdate(condType string, status me
 	})
 
 	if err := r.Status().Update(ctx, nodesensor); err != nil {
```
Comment (isimluk) [inline controllers/falcon/falconnodesensor_controller.go:772]: How come we are seeing these? In other words, who else is updating falconnodesensor resource behind our back?

Comment (redhatrises) [reply controllers/falcon/falconnodesensor_controller.go:772]: The error appears when an object which is sent for reconciliation while the previous updating is still in progress and can happen when the apiserver is getting too many requests so it is ratelimiting them. See https://github.com/kubernetes/kubernetes/issues/28149

I almost would feel like we should not ignore the error in the code and just document that if it happens/exists to safely ignore the error.

**Comment by redhatrises** [comment]
> I am trying to test this and I am getting:
> 
> ```
> 1.6566725897840126e+09  INFO    Waiting for cleanup pods to complete. Retrying....      {"controller": "falconnodesensor", "controllerGroup": "falcon.crowdstrike.com", "controllerKind": "FalconNodeSensor", "falconNodeSensor": {"name":"default"}, "namespace": "", "name": "default", "reconcileID": "c675d42a-c41c-4454-b9f1-5ec989b3e112", "DaemonSet": "/default"}
> ```
> 
> in an infinite loop. Is that something that can be related to this PR?

Not sure how it would be related at all. Is the daemonset-cleanup hung up on a pod?

**Comment by isimluk** [comment]
> Not sure how it would be related at all. Is the daemonset-cleanup hung up on a pod?

Right. This issue is not related to this PR.

File: controllers/falcon/falconnodesensor_controller.go (hunk 1 above)
Comment (isimluk) [reply controllers/falcon/falconnodesensor_controller.go:772]: I have spent some time looking further into this and I came to a conclusion that we indeed are causing this error (almost consistently) by a slight mis-use of the API. What I found is that we may update the nodesensor resource twice without fetching it.

Also, we actively set the condition at least 11 times per roll-out. I think if just didn't pushed the update on a condition, that already has a desired state, we may very well never see the error again.

**Comment by redhatrises** [comment]
Closing as this isn't a desired solution.

**Pull request closed without merging**